src/datasets/containers.jl

```diff
@@ -46,4 +48,36 @@ isimagefile(file::String) = occursin(IMAGEFILE_REGEX, lowercase(file))
 const IMAGEFILE_REGEX = r"\.(gif|jpe?g|tiff?|png|webp|bmp)$"
 
 
-## TODO: TableDataset
+struct TableDataset{T}
+    table::T #Should implement Tables.jl interface
+    TableDataset{T}(table::T) where T = Tables.istable(table) ? new{T}(table) : error("Object doesn't implement Tables.jl interface")
+end
+
+TableDataset(path::String) = TableDataset(DataFrame(CSV.File(path)))
+TableDataset(table::T) where {T} = TableDataset{T}(table)
+
+function LearnBase.getobs(dataset::FastAI.Datasets.TableDataset{T}, idx) where {T}
+    if Tables.rowaccess(dataset.table)
+        for (index, row) in enumerate(Tables.rows(dataset.table))
+            if index==idx
+                return row
+            end
+        end
```

Comment:
Iterating all the rows seems like a really unfortunate consequence of the Tables.jl interface. CSV.jl support "jumping ahead" to a certain row in the file. Does Tables.jl not have a similar interface?

Reply:
I don't think Tables.jl requires to support "jumping ahead" to a certain row, unless it is possible to do so for all objects implementing iteration interface.

If the table is row access, the only required method is `Tables.rows` which just needs to return an iterable of objects satisfying the `AbstractRow` interface. `AbstractRow` only requires methods for getting column values and names.

If the table is column access, the only required method is `Tables.columns` which needs to return an object implementing the `AbstractColumns` interface and having methods for getting a column object and column names.